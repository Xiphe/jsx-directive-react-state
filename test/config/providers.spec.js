/* eslint-disable camelcase */
const {
  registerProvider,
  __reset,
  getProviders,
} = require('../../config/providers');

describe('providers', () => {
  afterEach(() => {
    __reset();
  });

  it('has no registered providers by default', () => {
    expect(getProviders()).toEqual({});
  });

  it('provides registered providers', () => {
    registerProvider('foo', 'bar');

    expect(getProviders()).toEqual({ foo: 'bar' });
  });

  it('does not support overwrites of providers', () => {
    registerProvider('foo', 'bar');
    expect(() => {
      registerProvider('foo', 'bar');
    }).toThrowErrorMatchingSnapshot();
  });
});
