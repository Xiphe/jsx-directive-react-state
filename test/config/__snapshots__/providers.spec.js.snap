// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`providers does not support overwrites of providers 1`] = `"state provider \\"foo\\" is already registered"`;
